edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test/MerticsSerializerTests.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Runtime.InteropServices;
+    using System.Text;
+    using Microsoft.Azure.Devices.Edge.Util.Test.Common;
+    using Xunit;
+
+    public class MerticsSerializerTests
+    {
+        readonly Random rand = new Random();
+
+        [Fact]
+        public void SingleMetricSerializes()
+        {
+            Metric testMetric = new Metric(DateTime.UtcNow, this.metrics[0].name, this.rand.NextDouble() * 50, this.metrics[0].tags);
+
+            byte[] data = RawMetric.MetricsToBytes(new Metric[] { testMetric }).ToArray();
+            var reconstructedValues = RawMetric.BytesToMetrics(data).ToArray();
+
+            Assert.Single(reconstructedValues);
+            TestUtilities.ReflectionEqual(testMetric, reconstructedValues[0]);
+        }
+
+        [Fact]
+        public void SingleMetricSeriesSerializes()
+        {
+            var testMetrics = this.GenerateSeries(this.metrics[0].name, this.metrics[0].tags).ToArray();
+
+            byte[] data = RawMetric.MetricsToBytes(testMetrics).ToArray();
+            var reconstructedValues = RawMetric.BytesToMetrics(data).ToArray();
+
+            var expected = testMetrics.OrderBy(m => m.Name).ThenBy(m => m.Tags).ThenBy(m => m.TimeGeneratedUtc);
+            var actual = reconstructedValues.OrderBy(m => m.Name).ThenBy(m => m.Tags).ThenBy(m => m.TimeGeneratedUtc);
+            TestUtilities.ReflectionEqualEnumerable(expected, actual);
```

Comment:
Once IEquatable is implemented, you can just use Assert.Equal.

Reply:
Repeating comment above:
I would rather not implement a trait only for testing. It adds complexity to production code that isn't needed.